src/tools/compiletest/src/header.rs

```diff
@@ -525,17 +526,37 @@ impl TestProps {
                 self.check_test_line_numbers_match = config.parse_check_test_line_numbers_match(ln);
             }
 
-            if !self.run_pass {
-                self.run_pass = config.parse_run_pass(ln);
-            }
-
-            if !self.compile_pass {
-                // run-pass implies compile_pass
-                self.compile_pass = config.parse_compile_pass(ln) || self.run_pass;
-            }
-
-            if !self.skip_codegen {
-                self.skip_codegen = config.parse_skip_codegen(ln);
+            let check_no_run = |s| {
```

Comment:
This logic (down to the `match`) is sufficiently large and standalone that it would be good to extract it to a separate function.